pkg/windows/windows.go

```diff
@@ -335,6 +344,203 @@ func (vm *windows) Configure() error {
+	return u.Hostname(), nil
+}
+
+// resolveIpAddresses returns a list of strings representing the IP addresses
+// associated with the given hostname. If noHostsFile==true the entries in the
+// Windows hosts file are skipped when resolving the DNS query
+func (vm *windows) resolveIpAddresses(hostname string, noHostsFile bool) ([]string, error) {
+	// log info
+	vm.log.Info("Resolving IP Address(es)",
+		"hostname", hostname,
+		"noHostsFile", noHostsFile)
+	// define param
+	var noHostsFileParam = ""
+	if noHostsFile {
+		// enable flag
+		noHostsFileParam = "-NoHostsFile"
+	}
+	// resolve the IP address(es) for hostname
+	out, err := vm.Run("\"Resolve-DnsName "+
+		"-Name "+hostname+" "+
+		// ignore hosts file entries if needed
+		noHostsFileParam+" "+
+		// returns both A(IPv4) and AAAA(IPv6) records
+		"-Type A_AAAA "+
+		"| % IPAddress\"", true)
+	var ipAddresses []string
+	// check err
+	if err != nil {
+		// check not found
+		notFoundPrefix := fmt.Sprintf("%s : %s : %s",
+			"Resolve-DnsName", hostname, "DNS name does not exist")
+		if strings.HasPrefix(out, notFoundPrefix) {
+			// hostname not found, return empty list and no error
+			return ipAddresses, nil
+		}
+		// unknown error, return
+		return nil, err
+	}
+	// trim leading whitespaces and trailing CR LF (\r\n)
+	out = strings.TrimSpace(out)
+	// parse, where the separator is the new line escape sequence
+	// in Windows. See https://en.wikipedia.org/wiki/Newline#Representation
+	tokens := strings.Split(out, "\r\n")
+	// loop, cleanup and collect
+	for _, ip := range tokens {
+		// trim
+		ip = strings.TrimSpace(ip)
+		// check empty values
+		if ip == "" {
+			// skip
+			continue
+		}
+		// collect
+		ipAddresses = append(ipAddresses, ip)
+	}
+	// return
+	return ipAddresses, nil
+}
+
+// getApiServerInternalIpForPlatformStatus returns the APIServerInternalIP given the platform type
+//
+// Currently, VSphere is the only supported platform with the explicit APIServerInternalIP field,
+// as WMCO becomes available to other platforms (BareMetal, OpenStack, Ovirt, Kubevirt) support should be enabled
+// https://docs.openshift.com/container-platform/4.7/rest_api/config_apis/infrastructure-config-openshift-io-v1.html
+func getApiServerInternalIpForPlatformStatus(platformStatus *config.PlatformStatus) (string, error) {
+	if platformStatus == nil {
+		return "", errors.New("platformStatus cannot be nil")
+	}
+	switch platformStatus.Type {
+	case config.VSpherePlatformType:
+		return platformStatus.VSphere.APIServerInternalIP, nil
+		// TODO: uncomment as platforms become supported
+		// case config.BareMetalPlatformType:
+		// 	return platformStatus.BareMetal.APIServerInternalIP, nil
+		// case config.OpenStackPlatformType:
+		// 	return platformStatus.OpenStack.APIServerInternalIP, nil
+		// case config.OvirtPlatformType:
+		// 	return platformStatus.Ovirt.APIServerInternalIP, nil
+		// case config.KubevirtPlatformType:
+		// 	return platformStatus.Kubevirt.APIServerInternalIP, nil
```

Comment:
[YAGNI](https://martinfowler.com/bliki/Yagni.html) Alert!

the `APIServerInternalIP` is available for all of the above `platformStatus.Type`s, currently only `VSphere` is supported in WMCO. 

Happy to remove the comments. Just wanted to run this by the team.